project/webapp/controller/ProductsList.controller.js

```diff
@@ -1,15 +1,193 @@
+              })
+            );
+          });
+        }
+
+        return aFilters.length ? aFilters : null;
+      },
+
+      _getDateFilter() {
+        const sDate = this.byId("releaseDate");
+        const sDateStart = sDate.getDateValue();
+        const sDateEnd = sDate.getSecondDateValue();
+
+        if (!sDateStart || !sDateEnd) {
+          return null;
+        }
+
+        return new Filter({
+          path: "releaseDate",
+          operator: FilterOperator.BT,
+          value1: sDateStart.toISOString(),
+          value2: sDateEnd.toISOString(),
         });
+      },
+
+      _getCurrTokens(oEvent) {
+        let aSuppliersTokens = this.byId("supplier").getTokens();
+        const oMultiInput = oEvent.getParameters();
+        const sTokenType = oMultiInput.type;
+
+        if (sTokenType === this.TOKEN_REMOVED_TYPE) {
+          aSuppliersTokens = aSuppliersTokens.filter((item) => item !== oMultiInput.removedTokens[0]);
+        }
+
+        return aSuppliersTokens;
+      },
+
+      _getSupplierFilterWithoutToken() {
+        const sSupplierValue = this.byId("supplier").getValue();
+
+        return [
+          new Filter({
+            path: "suppliers",
+            operator: FilterOperator.EQ,
+            value1: sSupplierValue,
+            test: () => sSupplierValue === "",
+          }),
+        ];
+      },
+
+      _getSupplierFilterWithTokens(aSuppliersTokens) {
+        const aFilters = [];
+
+        aSuppliersTokens.forEach((sToken) => {
+          aFilters.push(
+            new Filter({
+              path: "suppliers",
+              operator: FilterOperator.EQ,
+              value1: sToken,
+              test: (aSuppliers) => {
+                const aResult = aSuppliers.filter(({id}) => id === sToken.getProperty("key"));
+
+                return !!aResult.length;
+              },
+            })
+          );
+        });
+
+        return aFilters;
+      },
+
+      _getSupplierFilter(oEvent) {
+        const aSuppliersTokens = this._getCurrTokens(oEvent);
+
+        if (!aSuppliersTokens.length) {
+          return this._getSupplierFilterWithoutToken(oEvent);
+        } else {
+          return this._getSupplierFilterWithTokens(aSuppliersTokens);
+        }
+      },
+
+      _getAllFilters(oEvent) {
+        let aFilters = [];
+        const categoriesFilter = this._getCategoriesFilter();
+        const suppliersFilter = this._getSupplierFilter(oEvent);
+
+        aFilters.push(this._getSearchNameFilter());
+        aFilters.push(this._getDateFilter());
+
+        if (categoriesFilter) {
+          aFilters.push(
+            new Filter({
+              filters: categoriesFilter,
+              and: false,
+            })
+          );
+        }
+        if (suppliersFilter) {
+          aFilters.push(
+            new Filter({
+              filters: suppliersFilter,
+              and: false,
+            })
+          );
+        }
+
+        aFilters = aFilters.filter((item) => {
```

Comment:
let's adjust logic to do not push nulls to aFilters, then we don't need to filter them out